### PR TITLE
add a polling manager with async mode

### PR DESCRIPTION
**Pull request by jsconan**
Polling manager: execute action at interval with async mode.

Each iteration wait for the previous one to be executed.

### CODE REVIEW

File: views/js/core/polling.js [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    var _defaultInterval = 60 * 1000;
+
+    /**
+     * Create a polling manager for a particular action
+     * @param {Object|Function} [config] - A config object, or the action called on each iteration
+     * @param {Function} [config.action] - The callback action called on each iteration
+     * @param {Number|String} [config.interval] - The minimal time between two iterations
+     * @param {Boolean} [config.autoStart] - Whether or not the polling should start immediately
+     * @param {Object} [config.context] - An optional context to apply on each action call
+     * @param {Function} [config.oneventname] - Bind an event handler to the event eventname
```
Comment (krampstudio) [inline views/js/core/polling.js]: Please do not attach listeners using the config

File: views/js/core/polling.js [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        /**
+         * Runs an iteration of the polling loop
+         */
+        var iteration = function iteration() {
+            /**
+             * Notifies the action is about to be called
+             * @event polling#call
+             * @param {polling} polling
+             */
+            polling.trigger('call', polling);
```
Comment (krampstudio) [inline views/js/core/polling.js]: Events doesn't need the context in parameter.

File: views/js/core/polling.js [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+             */
+            async : function async() {
+                var cb = {};
+
+                promise = new Promise(function(resolve, reject) {
+                    cb.resolve = resolve;
+                    cb.reject = reject;
+                });
+
+                _.assign(promise, cb);
```
Comment (krampstudio) [inline views/js/core/polling.js]: I think  you can just return the promise instead : 

```js
async : function async() {
   promise = new Promise();
   this.trigger('async', promise);
   return promise;
}
```

Or did I miss something ?

File: views/js/core/polling.js [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Jean-Sébastien Conan <jean-sebastien.conan@vesperiagroup.com>
```
Comment (krampstudio) [inline views/js/core/polling.js]: a little `@example` would be welcomed

File: views/js/core/polling.js (hunk 3 above)
Comment (jsconan) [inline views/js/core/polling.js]: Ok, maybe it is better to get access to the Promise object to add some actions. I agree for that, and it is already done.

However, the Promise object does not provide access to `resolve()`/`reject()` methods. So I still need to assign them into the final object. Natively, the only provided methods are `then()` and `catch()`

**Comment by jsconan** [comment]
Changes done.
Thanks for your review.

File: views/js/core/polling.js (hunk 3 above)
Comment (krampstudio) [inline views/js/core/polling.js]: Yes, my bad, the only way to resolve a promise is `Promise.resolve(promise)` which cannot be used here

**Pull request merged**